src/cluster/placement/compress.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package placement
+
+import (
+	"bytes"
+	"compress/flate"
```

Comment:
can we use https://github.com/klauspost/compress instead?

Reply:
Could you please elaborate why? :)

Reply:
more flexible api's (more interfaces generated for you), faster, drop-in replacement

Reply:
I updated benchmark test to use large simulated randomized placement:
```
pkg: github.com/m3db/m3/src/cluster/placement
BenchmarkCompressPlacementProto-12      	      10	 107019136 ns/op	         3.76 compress_rate/op
BenchmarkDecompressPlacementProto-12    	     171	   6995907 ns/op
```

Given how insignificant is compression/decompression overhead time, I would rather stick to standard library for the sake of stability.

Reply:
switched to that library and using `zstd` - it is indeed slightly better, thanks for suggestion